color-string.js

```diff
@@ -136,9 +145,16 @@ function getAlpha(string) {
 }
 
 // generators
-function hexString(rgb) {
-   return "#" + hexDouble(rgb[0]) + hexDouble(rgb[1])
-              + hexDouble(rgb[2]);
+function hexString(rgba, a) {
+   var a = (a !== undefined && rgba.length === 3) ? a : rgba[3];
+   return "#" + hexDouble(rgba[0]) 
+              + hexDouble(rgba[1])
+              + hexDouble(rgba[2])
+              + (
+                 (a >= 0 && a < 1)
```

Comment:
I think `a === 1` will cause a problem here

Reply:
I think it's fine because if `a === 1` (opaque), the alpha component is removed from the hex color (`#RRGGBB` instead of `#RRGGBBAA`).

Reply:
If `a === 1`, then there is no transparency, so I think we don't need to set the `alpha` component. Returning only a color formatted as `#rrggbb` would be a bit faster for the following operations.

Reply:
@etimberg What do you think ?